plugins/ptp_operator/metrics/logparser.go

```diff
@@ -486,32 +494,29 @@ func (p *PTPEventManager) ParseGNSSLogs(processName, configName, output string,
 	//openshift_ptp_offset_ns{from="gnss",iface="ens2f1",node="cnfde21.ptp.lab.eng.bos.redhat.com",process="gnss"} 0
 	if err == nil {
 		alias := getAlias(*iface)
-		lastState, errState := ptpStats[master].GetStateState(processName)
+		// last state of GNSS
+		lastState, errState := ptpStats[ifaceType].GetStateState(processName, iface)
 		pLabels := map[string]string{"from": processName, "node": ptpNodeName,
 			"process": processName, "iface": alias}
 		PtpOffset.With(pLabels).Set(gnssOffset)
 		SyncState.With(map[string]string{"process": processName, "node": ptpNodeName, "iface": alias}).Set(GetSyncStateID(syncState))
-		ptpStats[master].SetPtpDependentEventState(event.ClockState{
+		ptpStats[ifaceType].SetPtpDependentEventState(event.ClockState{
 			State:       GetSyncState(syncState),
 			Offset:      pointer.Float64(gnssOffset),
 			Process:     processName,
 			IFace:       iface,
 			Value:       map[string]int64{"gnss_status": gnssState},
 			ClockSource: GNSS,
 			NodeName:    ptpNodeName,
-			HelpText:    "0=NOFIX, 1=Dead Reckoning Only, 2=2D-FIX, 3=3D-FIX, 4=GPS+dead reckoning fix, 5=Time only fix",
+			HelpText:    map[string]string{"gnss_status": "0=NOFIX, 1=Dead Reckoning Only, 2=2D-FIX, 3=3D-FIX, 4=GPS+dead reckoning fix, 5=Time only fix"},
 		})
+		// reduce noise ; if state changed then send events
 		if lastState != GetSyncState(syncState) || errState != nil {
 			log.Infof("%s last state %s and current state %s", processName, lastState, GetSyncState(syncState))
-			var masterAlias string
-			if m, ok := ptpStats[master]; ok {
-				masterAlias = m.Alias()
-			}
-			if alias == "" {
-				masterAlias, _ = ptp4lCfg.GetUnknownAlias()
+			masterResource := fmt.Sprintf("%s/%s", alias, MasterClockType)
+			if !p.mock { /// to enable unit testing
```

Comment:
remove this, the unit test is actually checking if the event is sent.

Reply:
fixed it,